ingestion/src/kvkafka.rs

```diff
@@ -0,0 +1,321 @@
+use chrono::{DateTime, NaiveDateTime, Utc};
+use kafka::consumer::{Consumer, FetchOffset, GroupOffsetStorage};
+use serde::{Deserialize, Deserializer};
+use thiserror::Error;
+use tokio::sync::mpsc;
+
+use crate::PgConnectionPool;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("parsing xml error: {0}")]
+    IssueParsingXML(String),
+    #[error("parsing time error: {0}")]
+    IssueParsingTime(#[from] chrono::ParseError),
+    #[error("kafka returned an error: {0}")]
+    Kafka(#[from] kafka::Error),
+    #[error("postgres returned an error: {0}")]
+    Database(#[from] tokio_postgres::Error),
+    #[error(
+        "no Timeseries ID found for this data - station {}, param {}",
+        station,
+        param
+    )]
+    TimeseriesMissing { station: i32, param: i32 },
+    #[error("error while deserializing message: {0}")]
+    Deserialize(#[from] quick_xml::DeError),
+}
+
+#[derive(Debug, Deserialize)]
+/// Represents <KvalobsData>...</KvalobsData>
+struct KvalobsData {
+    station: Vec<Stations>,
+}
+#[derive(Debug, Deserialize)]
+/// Represents <station>...</station>
+struct Stations {
+    #[serde(rename = "@val")]
+    val: i32,
+    typeid: Vec<Typeid>,
+}
+#[derive(Debug, Deserialize)]
+/// Represents <typeid>...</typeid>
+struct Typeid {
+    #[serde(rename = "@val")]
+    val: i32,
+    obstime: Vec<Obstime>,
+}
+#[derive(Debug, Deserialize)]
+/// Represents <obstime>...</obstime>
+struct Obstime {
+    #[serde(rename = "@val")]
+    val: String, // avoiding parsing time at this point...
+    tbtime: Vec<Tbtime>,
+}
+#[derive(Debug, Deserialize)]
+/// Represents <tbtime>...</tbtime>
+struct Tbtime {
+    #[serde(rename = "@val")]
+    _val: String, // avoiding parsing time at this point...
+    _kvtextdata: Option<Vec<Kvtextdata>>,
+    sensor: Vec<Sensor>,
```

Comment:
The could would maybe be clearer to read if we named these vec fields as plurals? So `sensors` in this case. I guess it will probably require a rename annotation